contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -0,0 +1,319 @@
+        stabilityPool = IStabilityPool(_stabilityPool);
+        lqty = IERC20(_lqty);
+
+        if (!underlying.approve(_stabilityPool, type(uint256).max)) {
+            revert TokenApprovalFailed(_underlying);
+        }
+    }
+
+    /**
+     * Transfers administrator rights for the Strategy to another account,
+     * revoking current admin roles and setting up the roles for the new admin.
+     *
+     * @notice Can only be called by the account with the ADMIN role.
+     *
+     * @param _newAdmin The new Strategy admin account.
+     */
+    function transferAdminRights(address _newAdmin) external onlyAdmin {
+        if (_newAdmin == address(0x0)) revert StrategyAdminCannotBe0Address();
+
+        if (_newAdmin == msg.sender)
+            revert StrategyCannotTransferAdminRightsToSelf();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newAdmin);
+        _setupRole(MANAGER_ROLE, _newAdmin);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _revokeRole(MANAGER_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Returns true since strategy is synchronous.
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return true;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return investedAssets() > 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice ETH & LQTY rewards of the strategy waiting to be claimed in the liquity stability pool are not included
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return stabilityPool.getCompoundedLUSDDeposit(address(this));
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        stabilityPool.provideToSP(balance, address(0));
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice will also claim unclaimed LQTY & ETH gains
+    /// @notice when amount > total deposited, all available funds will be withdrawn
+    function withdrawToVault(uint256 amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (amount == 0) revert StrategyAmountZero();
+        if (amount > investedAssets()) revert StrategyNotEnoughShares();
+
+        // withdraws underlying amount and claims LQTY & ETH rewards
+        stabilityPool.withdrawFromSP(amount);
+
+        uint256 lqtyRewards = lqty.balanceOf(address(this));
+        uint256 ethRewards = address(this).balance;
+        // backend to keep track of this and decide when to call reinvestRewards
+        emit StrategyRewardsClaimed(lqtyRewards, ethRewards);
```

Comment:
I'm confused. The backend will keep track of this, but why are we emitting it here? It's not like this operation changes the rewards, right?

Reply:
no, it doesn't change, it just claims the rewards. The backend will call the `harvest `method to collect rewards, to my understanding this should happen a few times in a period of one week for example. However, rewards are also collected when `withdrawToVault `is called, so if there are multiple withdrawals that will claim the rewards between two calls to `harvest`, maybe we want to have some kind of threshold on the backend side that will track how much rewards have been accumulated and decide to call `reinvestRewards` if the threshold has been met, before a "scheduled" call to `harvest`.

Reply:
I think I understand, but the backend could check the contract directly to make that decision. And information coming from the contract is fresher than information coming from the subgraph

Reply:
Thats a really good point. Will help us save gas in this withdrawToVault method too. I think we can remove this event emission. @brkomir what do you think?

Reply:
Yes, the backend can fetch fresh information from the contract, by checking LQTY and ETH balances of the strategy, and also checking unclaimed LQTY and ETH rewards from the stability pool contract.
I'm not sure if we should not have this event at all, or only have the event emitted without any parameters? Just so the backend can be informed when rewards are claimed outside the `harvest` method

Reply:
Looking at this again, you may want to keep the event as it is. The idea is to allow programs reading these events to recreate the internal state, so they would need this information. I would remove the comment though

Reply:
[removed](https://github.com/lindy-labs/sc_solidity-contracts/pull/171/commits/5dc356c2ae4d2f7cdc4083c4f66c7dd4b14af6c1)